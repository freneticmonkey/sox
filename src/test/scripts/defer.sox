fun hello(name) {

    defer () {
        print "sleep tight";
    }
    
    defer () {
        var text = "goodbye: " + name;
        print text;
    }

    print "hello: " + name;

    
    usleep(100);
}

var start = clock();
hello("world");
print clock() - start;