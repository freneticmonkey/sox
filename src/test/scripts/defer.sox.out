hello: world
sleep tight
goodbye: world
